pkg/auth/metrics.go

```diff
@@ -79,10 +80,17 @@ func (m *Metrics) loginSuccessfulSync(k8sConfig *rest.Config, ls *sessions.Login
 		return
 	}
 
+	anonymousInternalProxiedK8SRT, err := rest.TransportFor(rest.AnonymousClientConfig(m.internalproxyClientConfig))
+	if err != nil {
+		klog.Errorf("failed to set up an anonymous roundtripper: %v", err)
+		return
+	}
+
 	ctx := context.TODO()
+
 	configWithBearerToken := &rest.Config{
 		Host:        k8sConfig.Host,
-		Transport:   k8sConfig.Transport,
```

Comment:
`k8sConfig.Transport` is nil.

This the entire k8sConfig:

```
I0828 11:02:11.879264 1 metrics.go:114] auth.Metrics loginSuccessfulSync - k8sConfig: &rest.Config{Host:"https://kubernetes.default.svc", APIPath:"", ContentConfig:rest.ContentConfig{AcceptContentTypes:"", ContentType:"", GroupVersion:(*schema.GroupVersion)(nil), NegotiatedSerializer:runtime.NegotiatedSerializer(nil)}, Username:"", Password:"", BearerToken:"", BearerTokenFile:"/var/run/secrets/kubernetes.io/serviceaccount/token", Impersonate:rest.ImpersonationConfig{UserName:"", UID:"", Groups:[]string(nil), Extra:map[string][]string(nil)}, AuthProvider:<nil>, AuthConfigPersister:rest.AuthProviderConfigPersister(nil), ExecProvider:<nil>, TLSClientConfig:rest.sanitizedTLSClientConfig{Insecure:false, ServerName:"", CertFile:"", KeyFile:"", CAFile:"/var/run/secrets/kubernetes.io/serviceaccount/ca.crt", CertData:[]uint8(nil), KeyData:[]uint8(nil), CAData:[]uint8(nil), NextProtos:[]string(nil)}, UserAgent:"", DisableCompression:false, Transport:http.RoundTripper(nil), WrapTransport:(transport.WrapperFunc)...
```

From here, you can see that `Transport:http.RoundTripper(nil)`

Therefore we needed to create the Transport.

Reply:
lets pass the already instantiated `anonymousInternalProxiedK8SRT` from the server.go